indexer/reindex_bags_backlog.py

```diff
@@ -0,0 +1,191 @@
+
+        if(version > stored_version):
+            bags[dynamo_id] = version
+
+    print(f"latest_bags: {len(bags)}")
+
+    return bags
+
+
+def publish_bags(sns_client, topic_arn, bags):
+    published_bags = []
+    unique_bags = len(bags)
+
+    for (dynamo_id, version) in bags.items():
+        space, external_id = dynamo_id.split("/", 1)
+        published_bags.append(dynamo_id)
+
+        payload = fake_known_replicas_payload(space, external_id, version)
+        #publish_payload(sns_client, topic_arn,  payload)
+        publish_count = len(published_bags)
+
+        if(publish_count % 10 == 0 or publish_count > (unique_bags - 10)):
+            print(f"{publish_count}/{unique_bags}")
+
+    print(f"published_bags: {len(published_bags)}")
+
+    return published_bags
+
+def confirm_indexed(elastic_client, published_bags):
+    def _chunks(big_list, chunk_length):
+        for i in range(0, len(big_list), chunk_length):
+            yield big_list[i:i + chunk_length]
+
+    def _query(ids):
+        query_body = {
+            "query": {
+                "ids" : {
+                    "values" : ids
+                }
+            }
+        }
+
+        s = Search(index="storage_stage_bags").using(elastic_client).update_from_dict(query_body)
+
+        found_ids = [hit.id for hit in s.scan()]
+
+        return set(ids).difference(found_ids)
+
+    diff_list = [_query(chunk) for chunk in _chunks(published_bags, 50)]
+    flat_list = [item for sublist in diff_list for item in sublist]
+
+    return flat_list
+
+def create_elastic_client(role_arn, es_secrets):
+    secretsmanager_client = create_client("secretsmanager", role_arn)
+
+    def _get_secret(secret_id):
+        response = secretsmanager_client.get_secret_value(SecretId=secret_id)
+
+        return response['SecretString']
+
+    config = { key:_get_secret(value) for (key, value) in es_secrets.items() }
+
+    return Elasticsearch(
+        [config['hostname']],
+        http_auth=(config['username'], config['password']),
+        scheme="https",
+        port=9243,
+    )
+
+if __name__ == "__main__":
+    role_arn = 'arn:aws:iam::975596993436:role/storage-developer'
+    table_name = 'vhs-storage-staging-manifests'
+    #table_name = 'vhs-storage-manifests'
+    topic_arn = 'arn:aws:sns:eu-west-1:975596993436:storage_staging_bag_register_output'
+
+    es_secrets = {
+        'username': 'storage_bags_reindex_script/es_username',
+        'password': 'storage_bags_reindex_script/es_password',
+        'hostname': 'storage_bags_reindex_script/es_hostname',
+    }
+
+    dynamodb_client = create_client("dynamodb", role_arn)
+    sns_client = create_client("sns", role_arn)
+    elastic_client = create_elastic_client(role_arn, es_secrets)
+
+    bags_to_publish = get_latest_bags(dynamodb_client, table_name)
+    published_bags = publish_bags(sns_client, topic_arn, bags_to_publish)
+    not_indexed = confirm_indexed(elastic_client, published_bags)
```

Comment:
Should the "confirm indexed" stuff live in a different script? Publishing everything to SNS feels like it'll be faster than getting it indexed.

If you do leave it here, include some context for the number you're printing below.

Reply:
Going to make this use click and split commands that way.